src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/services/StaticInvalidationService.kt

```diff
@@ -0,0 +1,156 @@
+package nl.tudelft.ewi.se.ciselab.testgenie.services
+
+import com.intellij.openapi.editor.Document
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiCodeBlock
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiModifierList
+import com.intellij.psi.PsiReferenceParameterList
+import com.intellij.psi.PsiWhiteSpace
+import com.intellij.refactoring.suggested.endOffset
+import com.intellij.refactoring.suggested.startOffset
+
+class StaticInvalidationService {
+
+    // HashMap of Class filePath -> (HashMap of Method Signature -> (Method Body, Covered Lines))
+    private var savedMethods: HashMap<String, HashMap<String, Pair<ArrayList<PsiElement>, Set<Int>>>> = HashMap()
+
+    /**
+     * Checks if method has been changed since last test generation (no, if first run)
+     * @param signature the method in question
+     * @param body list of the body-elements of the method, without whitespaces
+     * @param methods hashmap of previously tested methods and their bodies
+     * @return the lines which should be deleted (based on previous lines for method)
+     */
+    private fun validateMethod(
```

Comment:
Private methods are usually placed below public ones

Reply:
will be addressed in code

Reply:
addressed